frontend/packages/dev-console/src/utils/shared-submit-utils.ts

```diff
@@ -43,8 +43,8 @@ export const createService = (
 
   let ports = imagePorts;
   const buildStrategy = formData.build?.strategy;
-  if (buildStrategy === 'Docker') {
-    const port = { containerPort: _.get(formData, 'docker.containerPort'), protocol: 'TCP' };
+  if (buildStrategy === 'Docker' && unknownTargetPort) {
+    const port = { containerPort: _.toInteger(unknownTargetPort), protocol: 'TCP' };
```

Comment:
Instead not adding a port at all maybe we can add the `defaultUnknownPort` here in case there is no `unknownTargetPort`

Reply:
Updated.